maintenance/wikia/HttpsMigration/migrateCustomJs.php

```diff
@@ -0,0 +1,350 @@
+				$url = $this->upgradeThirdPartyUrl( $url );
+			}
+		} elseif ( $this->isWikiaImageUrl( $url ) ) {
+			$url = $this->forceWikiaImageOverHttps( $url );
+			$this->logUrlChange( 'Replaced vignette url', $originalUrl, $url );
+		} elseif ( $this->isWikiaComSubdomainUrl( $url ) ) {
+			$url = $this->fixWikiUrl( $url );
+			$this->logUrlChange( 'Converted wiki url to protocol/host relative', $originalUrl, $url );
+		} else {
+			$this->output( "Don't know to handle {$url}\n" );
+		}
+		return $url;
+	}
+
+	/**
+	 * @param $matches - regex match array, $matches[1] contain the protocol and domain
+	 * @return updated url address match
+	 */
+	public function makeUrlHttpsComatible( $matches ) {
+		$this->output( "Updating '{$matches[1]}' used in {$matches[0]}\n" );
+		$domain = $this->fixUrl( $matches[1] );
+		if ( $domain != $matches[1] ) {
+			$this->output( "Updating '{$matches[1]}' to {$domain}\n" );
+			$matches[ 0 ] = str_replace( $matches[ 1 ], $domain, $matches[ 0 ] );
+		}
+		return $matches[ 0 ];
+	}
+
+	/**
+	 * Process JS source code upgrading urls to https/protocol-relative.
+	 * @param $text JS source code
+	 * @return mixed Updated JS source code
+	 */
+	public function updateJSContent( $text ) {
+		$lines = explode( "\n", $text );
+		foreach ( $lines as $index => $line ) {
+			// Try to skip the most common comment patterns
+			if ( !preg_match( '/^(\s*\*|\s*\/\/\s+)/', $line ) ) {
+				$lines[$index] = preg_replace_callback( '/(https?:\/\/[a-zA-Z0-9\\.]+)/i', [ $this, 'makeUrlHttpsComatible' ], $line );
+				// log urls that were not altered, most likely there will be a lot of those
+				if ( strpos( $lines[$index], 'http://' ) !== false ) {
+					$this->output( "Notice: http protocol used in \"{$lines[$index]}\"\n" );
+				}
+			}
+		}
+
+		$text = implode( "\n", $lines );
+		return $text;
+	}
+
+	/**
+	 * If possible, make the JS https-ready and save the updated content (if not running in dry mode)
+	 */
+	public function migrateJS( Title $title ) {
+		global $wgCityId, $wgRequest;
+		$revId = $title->getLatestRevID( Title::GAID_FOR_UPDATE );
+		$this->currentTitle = $title;
+		$revision = Revision::newFromId( $revId );
+		if( !is_null( $revision ) ) {
+			$text = $revision->getText();
+			$updatedText = $this->updateJSContent( $text );
+			if ($text !== $updatedText) {
+				if ( $this->saveChanges ) {
+					// pay attention to JS-review process. if the latest revision is the same as
+					// the one returned from ContentReviewHelper's getReviewedRevisionId, just autoapprove
+					// (which requires FANDOMBot to be added to the content review group.
+					// autoapprove can be set by using wpApprove parameter.
+					// if the current JS revision is not approved yet, do not autoapprove JS changes!
+					$latestReviewedID = $this->contentReview->getReviewedRevisionId( $title->getArticleID(), $wgCityId );
+					if ( $revId === $latestReviewedID ) {
+						$this->output( "Auto-approving change\n" );
```

Comment:
As discussed we should autoapprove images urls and go through review flow for other changes